vega_sim/scenario/common/agents.py

```diff
@@ -1638,7 +1641,16 @@ def _optimal_strategy(self, current_position, current_step):
             market_id=self.external_market_id
         )
 
-        int_fee = self.int_mkr_fee + self.int_liq_fee
+        if (ext_best_bid == 0) or (ext_best_ask == 0):
+            return None, None
+
+        fee_share = self.vega.get_liquidity_fee_shares(
+            market_id=self.market_id,
+            wallet_name=self.wallet_name,
+            key_name=self.key_name,
+        )
+
+        int_fee = (self.int_mkr_fee + self.int_liq_fee) * fee_share
```

Comment:
Maker fee should still go entirely to them (or not if they aren't the LP who is hit)